modules/register/commerce_pos_register.module

```diff
@@ -609,15 +609,23 @@ function commerce_pos_register_get_float($register_id, $date = FALSE) {
     $date = REQUEST_TIME;
   }
 
-  $date = format_date($date, 'custom', 'Y-m-d');
-
-  $result = db_query("SELECT float_id, register_id, amount, created
-    FROM {commerce_pos_register_float}
-    WHERE register_id = :register_id
-    AND DATE_FORMAT(FROM_UNIXTIME(created), '%Y-%m-%d') = :date", array(
-    ':register_id' => $register_id,
-    ':date' => $date,
-  ));
+  //Grab the beginning and end of the day
+  $begin_of_day = strtotime("midnight", $date);
```

Comment:
I'd go with "start_of_day" matched better with end and means you don't have to abbreviate beginning to begin